### ISSUE TITLE
GeneratorDatasetOp::Dataset will not be optimized because the dataset does not implement the AsGraphDefInternal() method needed to apply optimizations.

### ISSUE THREAD

**Issue by Dex247** [nicknochnack/TFODCourse#102]
I am trying to train my model. I am using a CPU machine, with 6GB Ram. Could it be the memory is not adequate for the model training? 

 ---- GeneratorDatasetOp::Dataset will not be optimized because the dataset does not implement the AsGraphDefInternal() method needed to apply optimizations.

How do rectify this error? @nicknochnack 

![image](https://user-images.githubusercontent.com/38394300/182932603-55a3e630-1f91-4203-aa59-26f96874d322.png)

**Comment by Yozor4** [comment]
have you fixed this error? I am having the same issue

**Comment by ASH-will-WIN** [comment]
I am also having a very similar error

**Comment by Dex247** [comment]
Yes I got it solved. Make your you have CUDA and CUDNN all in place. Use a
GPU machine and the problem will be solved.
